stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -385,12 +317,13 @@ size(C::Union{Cholesky, CholeskyPivoted}, d::Integer) = size(C.factors, d)
 function getproperty(C::Cholesky, d::Symbol)
     Cfactors = getfield(C, :factors)
     Cuplo    = getfield(C, :uplo)
+    info     = getfield(C, :info)
     if d == :U
-        return UpperTriangular(Symbol(Cuplo) == d ? Cfactors : copy(Cfactors'))
+        return @assertposdef UpperTriangular(Symbol(Cuplo) == d ? Cfactors : copy(Cfactors')) info
```

Comment:
The reason I added this last commit is that `chol(A)` used to error when the factorization failed. Now that `chol(A)` is replaced with `cholesky(A).U` we need this check here to keep the same behavior. If you really want to access the failed factor you can use `getfield(A, :U)`.

On that note, we should probably do the same thing for `lu(A).L` etc, although `lu(A)` happily returned before without checking if the factorization failed. Thoughts?

Reply:
I might err to following the existing `*fact` methods' behavior and forgoing these checks. Thoughts @andreasnoack?

Reply:
We should probably check. It is a bit unfortunate that expert users would need to use `getfield` but the alternative is that we'd have to teach all linear algebra users to always use `issuccess` before doing anything with factorizations.

Reply:
I assume that users call cholesky either: 1) knowing that their matrices are positive definite; or 2) not being certain that their matrices are positive definite, and then checking for success after the call. Otherwise I assume they would use another, more appropriate decomposition or, absent knowledge of which decomposition to use, call `factorize`? And as such not checking in `cholfact` seems to have been fine so far?